File: src/routes/auth.routes.js
import express from "express"

import * as signinControllers from "../controllers/auth/signin.controllers.js"
import * as passwordResetControllers from "../controllers/auth/passwordReset.controllers.js"
import * as signupControllers from "../controllers/auth/signup.controllers.js"
import * as signupValidators from "../validators/auth/signup.validators.js"
import * as signinValidators from "../validators/auth/signin.validators.js"
import * as passwordResetValidators from "../validators/auth/passwordReset.validators.js"
import { expressSession } from "../middlewares/auth.middlewares.js"

const router = express.Router()

router.use(expressSession())

router.post(
  "/signup/request_new_account",
  ...signupValidators.requestNewAccount,
  signupControllers.requestNewAccount
)
router.post(
  "/signup/verify_email",
  ...signupValidators.verifyEmail,
  signupControllers.verifyEmail
)
router.post(
  "/signup/register_user",
  ...signupValidators.registerUser,
  signupControllers.registerUser
)

router.post("/signin", ...signinValidators.signin, signinControllers.signin)

router.post(
  "/forgot_password/request_password_reset",
  ...passwordResetValidators.requestPasswordReset,
  passwordResetControllers.requestPasswordReset
)
router.post(
  "/forgot_password/confirm_email",
  ...passwordResetValidators.confirmEmail,
  passwordResetControllers.confirmEmail
)
router.post(
  "/forgot_password/reset_password",
  ...passwordResetValidators.resetPassword,
  passwordResetControllers.resetPassword
)

export default router

File: src/domain/auth/authRoutes/authRoutes.go
package authRoutes

import (
	"i9lyfe/src/domain/auth/authMiddlewares"
	"i9lyfe/src/domain/auth/login/loginControllers"
	"i9lyfe/src/domain/auth/passwordReset/passwordResetControllers"
	"i9lyfe/src/domain/auth/signup/signupControllers"

	"github.com/gofiber/fiber/v3"
)

func Routes(router fiber.Router) {
	router.Post(
		"/signup/request_new_account",
		signupControllers.RequestNewAccount,
	)

	router.Post(
		"/signup/verify_email",
		authMiddlewares.SignupSession,
		signupControllers.VerifyEmail,
	)

	router.Post(
		"/signup/register_user",
		authMiddlewares.SignupSession,
		signupControllers.RegisterUser,
	)

	router.Post(
		"/login",
		loginControllers.Login,
	)

	router.Post(
		"/forgot_password/request_password_reset",
		passwordResetControllers.RequestPasswordReset,
	)

	router.Post(
		"/forgot_password/confirm_email",
		authMiddlewares.PasswordResetSession,
		passwordResetControllers.ConfirmEmail,
	)

	router.Post(
		"/forgot_password/reset_password",
		authMiddlewares.PasswordResetSession,
		passwordResetControllers.ResetPassword,
	)
}
